### PR TITLE
vesktop: add darwin support

### PR DESCRIPTION
**Pull request by getchoo**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by getchoo** [comment]
@uncenter do you have `sandbox = true` set in nix.conf? my best guess here is that it can see you have some code signing certs available on your machine but nix can't access them during the build; other than that i'm not sure what's going on

ofborg for example builds this just fine: https://logs.ofborg.org/?key=nixos/nixpkgs.313797&attempt_id=a85c9296-0aea-499e-97e2-df30a2959a87

**Comment by uncenter** [comment]
> @uncenter do you have `sandbox = true` set in nix.conf? my best guess here is that it can see you have some code signing certs available on your machine but nix can't access them during the build; other than that i'm not sure what's going on
> 
> ofborg for example builds this just fine: [logs.ofborg.org/?key=nixos/nixpkgs.313797&attempt_id=a85c9296-0aea-499e-97e2-df30a2959a87](https://logs.ofborg.org/?key=nixos/nixpkgs.313797&attempt_id=a85c9296-0aea-499e-97e2-df30a2959a87)

Yeah I do have sandboxing on. Should I disable it? Not sure how to fix.

**Comment by getchoo** [comment]
i've explicitly disabled code signing with the `CSC_IDENTITY_AUTO_DISCOVERY` env var from https://www.electron.build/code-signing.html#how-to-disable-code-signing-during-the-build-process-on-macos, which *should* fix this

**Comment by uncenter** [comment]
Result of `nixpkgs-review pr 313797` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>vesktop</li>
  </ul>
</details>

File: pkgs/by-name/ve/vesktop/package.nix [hunk 1]
```diff
@@ -126,63 +144,81 @@ stdenv.mkDerivation (finalAttrs: {
...
-  # this is consistent with other nixpkgs electron packages and upstream, as far as I am aware
-  installPhase = ''
-    runHook preInstall
-
-    mkdir -p $out/opt/Vesktop
-    cp -r dist/linux-*unpacked/resources $out/opt/Vesktop/
-
+  postBuild = lib.optionalString stdenv.isLinux ''
     pushd build
     ${libicns}/bin/icns2png -x icon.icns
```
Comment (vgskye) [inline pkgs/by-name/ve/vesktop/package.nix:167]: can't we move this to installPhase?

Comment (getchoo) [reply pkgs/by-name/ve/vesktop/package.nix:167]: i would consider generating the icons to be building. i also just think it looks a bit cleaner

**Comment by getchoo** [comment]
@uncenter: it seems some [other packages] manually exported the env var, so i've done the same. if you have the time, it'd be awesome if you could test again :)

(hopefully this will be the last time)

**Review by SuperSandro2000 — APPROVED** [review]
LGTM

**Comment by uncenter** [comment]
Result of `nixpkgs-review pr 313797` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>vesktop</li>
  </ul>
</details>

**Comment by getchoo** [comment]
fixed merge conflict

**Comment by AVOnyx** [comment]
Result of `nixpkgs-review pr 313797` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vesktop</li>
  </ul>
</details>

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 313797` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vesktop</li>
  </ul>
</details>

**Pull request merged**